### PR TITLE
fix(locales): use correct lang

### PR DESCRIPTION
**Pull request by posva**
- Correctly changes the lang attribute when switching languages
- Removes the limitation of prod only lang attribute
- Allows multi languages searches with DocSearch

### CODE REVIEW

File: src/client/app/index.ts [hunk 1]
```diff
@@ -15,6 +22,20 @@ const NotFound = Theme.NotFound || (() => '404 Not Found')
...
   name: 'VitePressApp',
   setup() {
+    const siteData = useSiteDataByRoute()
+
+    // change the language on the HTML element based on the current lang
+    onMounted(() => {
+      watch(
+        () => siteData.value.lang,
+        (lang: string) => {
+          console.log('lang', lang)
```
Comment (kiaking) [inline src/client/app/index.ts:32]: Should we remove this 👀 ?

**Review by kiaking — COMMENTED** [review]
Niiiice! Just made a comment!

File: src/client/app/index.ts (hunk 1 above)
Comment (posva) [reply src/client/app/index.ts]: Yeah, thanks 😆

**Review by kiaking — APPROVED** [review]
:tada: :tada: :tada:

**Comment by posva** [comment]
BTW here is an example https://github.com/algolia/docsearch-configs/pull/3906

**Pull request merged**